### ISSUE TITLE
Converging slow

### ISSUE THREAD

**Issue by Yulongggggg** [pratikrathore8/opt_for_pinns#1]
Hi,

Do you know if there's any solution when converging is very slow?

11112,NNCG,4.061273e-03,1.327160e+00,8.344561e-02,-3.001276e+07,-1.424253e-01,1.522003e-03
11113,NNCG,4.061261e-03,1.327160e+00,8.274938e-02,-3.001276e+07,-1.422030e-01,1.522003e-03
11114,NNCG,4.061250e-03,1.327160e+00,8.206061e-02,-3.001276e+07,-1.419813e-01,1.522003e-03
11115,NNCG,4.061238e-03,1.327160e+00,8.137920e-02,-3.001276e+07,-1.417601e-01,1.522003e-03
11116,NNCG,4.061227e-03,1.327160e+00,8.070510e-02,-3.001276e+07,-1.415396e-01,1.522003e-03
11117,NNCG,4.061216e-03,1.327160e+00,8.003823e-02,-3.001276e+07,-1.413197e-01,1.522003e-03
11118,NNCG,4.061204e-03,1.327160e+00,7.937853e-02,-3.001276e+07,-1.411003e-01,1.522003e-03

**Comment by pratikrathore8** [comment]
Hi @Yulongggggg 

It's hard to say exactly what is going on without more information. I will warn you that NNCG is sensitive to the choice of hyperparameters.

In particular, the damping `mu` can really impact the performance. If `mu` is too small, NNCG can break down due to numerical stability issues; if `mu` is too large, NNCG will converge slowly because its behavior will imitate gradient descent.

Let me know if you need more help.
